### ISSUE TITLE
Backspace key in Note-input mode does not successively delete the notes being entered

### ISSUE THREAD

**Issue by LupusCaeruleus** [musescore/MuseScore#17557]
### Issue type

Other type of issue

### Bug description

When composing or transcribing a score, one will quite often want to delete the sequence of notes just entered because of a change of mind or an error.

In MuseScore 3.x, after entering Note-input mode, the user can press the Backspace key at any time to delete whatever was just entered successively in reverse order; and after exitting and re-entering Note-input mode, the user can still press the Backspace key to continue the last deleting succession. This is quite convenient and intuitive, as the Backspace key here functions similarly as almost any other apps like text editors.

However, in MuseScore 4.0.2, only the last note entered can be deleted no matter how many times the Backspace key is pressed in Note-input mode. The user will have to exit Note-input mode to select-delete the unwanted notes or press Ctrl+Z instead, making the same process more awkward and time-consuming. I would expect the Backspace key in MuseScore 4.x to behave similarly in Note-input mode as it does in MuseScore 3.x.

(See the screen recording below for comparison.)

### Steps to reproduce

1. Open or create any MuseScore 4.0.2 file.
2. Select any bar (measure) or any element, click the pencil icon on the top left or press N to enter Note-input mode.
3. Enter a sequence of notes (and/or anything else that can be directly entered in Note-input mode), then press the Backspace key for multiple times.
4. No matter how many times the Backspace key is pressed, only the last note entered will be deleted.

Expected behavior: The entered note sequence are to be deleted in succession in the reverse order every time the Backspace key is pressed, just as MuseScore 3.x does.

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/70313425/0463e37c-9210-4988-955d-6ca5bb2320c2

### MuseScore Version

4.0.2

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Windows 10

**Comment by shoogle** [comment]
For context, I believe in MuseScore 3 the Backspace key was mapped to the `undo` function in note-input mode, so pressing Backspace was literally equivalent to pressing Ctrl+Z (but only in note-input mode).

**Comment by MarcSabatella** [comment]
Indeed.  I was never a fan of that behavior, and the consensus was it would be better to implement a true backspace, but that hasn't happened, obviously - right now it's just delete.  See https://github.com/musescore/MuseScore/issues/9188

FWIW, I can't imagine a true backspace command would be hard to implement.  It's basically just the existing delete command followed by the existing select previous chord command.

**Comment by Jojo-1000** [comment]
> It's basically just the existing delete command followed by the existing select previous chord command.

I like the current behavior that chords (when typed using shift + letter) are deleted one note at a time as they were typed. It is easier than Ctrl+Z when mistyping. So it would be nice to preserve that and only move back one note if the current is empty.

**Comment by kurtmetzger** [comment]
So after doing some digging, it looks like in the "shortcuts.xml", the backspace and delete keys call on the same action:
`    <SC>
        <key>action://delete</key>
        <seq>Del</seq>
        <seq>Backspace</seq>
    </SC>`

For backspace to behave differently from delete in this instance, they would need to be separated here first, yes? Or is it possible/ make more sense to define different behaviors for them somewhere further down the line?

**Comment by rbbandeira** [comment]
i'm currently trying to fix this bug. can you assign it to me?

**Comment by yooyoo1024** [comment]
I have almost finished fixing it. I didn't think someone else would pick it up right now

**Comment by stevage** [comment]
No one seems to have mentioned this yet: this discussion seems to assume that backspace and del are separate keys available to the user. But on most Mac keyboards (laptops, and small external keyboards I think) there is only one: it's labelled "delete" but is in the backspace location, and behaves like backspace (deleting the character to the left of the caret, in text editing contexts). I think it sometimes looks like an X in a left pointing arrow. (As opposed to the other "delete" key which only exists on full size external keyboards, I think, which looks like an X in a right pointing arrow) (See [this](https://www.techfinitive.com/explainers/where-is-the-delete-key-on-a-mac-keyboard/) for a better explanation).

It seems clear that there should be separate commands for "delete" (remove the selected note) and "backspace" (remove the selected note, and then if there are no notes left at the current location, and there is a note immediately to the left, select that instead. (I think? Is this design in question?)

But what should the default be for the backspace/"delete" key be on a Mac? Make it behave like backspace, and users can remap it to the "delete" functionality if they don't like it?
